tezos/encoding/src/binary_writer.rs

```diff
@@ -226,7 +226,10 @@ fn encode_value(data: &mut Vec<u8>, value: &Value, encoding: &Encoding) -> Resul
             match value {
                 Value::Enum(_, ordinal) => {
                     // TODO: handle enum of different sizes
-                    data.put_u8(ordinal.expect("Was expecting enum ordinal value") as u8);
+                    data.put_u8(
+                        ordinal.ok_or_else(|| Error::custom("Was expecting enum ordinal value"))?
+                            as u8,
```

Comment:
Why is ordinal `u32` but being casted to `u8` ? Can we use `TryFrom` or something like that here?

Reply:
Good point. Let me check that.